src/fspec-archive.c

```diff
@@ -54,26 +37,29 @@ defaultmode(const char *type)
     exit(1);
 }
 
-int main(int argc, char **argv)
+int
+main(int argc, char **argv)
 {
     char *line = NULL;
     size_t len = 0;
     ssize_t n = 0;
+    int datafd = -1;
+    struct archive *a = NULL;
+    struct archive_entry *entry = NULL;
 
     a = archive_write_new();
     entry = archive_entry_new();
-    check(!!a);
-    check(!!entry);
+    assert(a);
+    assert(entry);
```

Comment:
`assert` should only be used for checking invariants indicating an internal error. If compiled with `-D NDEBUG`, the checks will get compiled out.